src/problem.ts

```diff
@@ -1,62 +1,78 @@
-
-
-    constructor(problem: ProblemInterface, customKeys?: string[]) {
-        super(problem.detail || problem.title);
-        this.http = problem.http
-        this.type = problem.type
-        this.title = problem.title;
-        this.detail = problem.detail;
-        this.instance = problem.instance;
-        this.stack = problem.stack;
-        customKeys?.map((customKey) => {
-            this[customKey] = problem[customKey];
-        })
-    }
+  public type: string;
+  public title: string;
+  public instance?: string;
+  public detail?: string;
+  public http?: httpObject;
+  [key: string]: any;
 
-   copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
-        switch (mode) {
-
-            case COPY_MODE.LEAVE_PROPS:
-                return new Problem(problem, props);
-
-            case COPY_MODE.SKIP_PROPS:
-            default:
-                let keysToBeCopied: string[] = [];
-                for (let key in problem) {
-                    if (props.includes(key))
-                        continue;
-                    keysToBeCopied.push(key)
-                }
-                return new Problem(problem, keysToBeCopied)
-        }
-    };
+  constructor(problem: ProblemInterface, customKeys?: string[]) {
+    super(problem.detail || problem.title);
+    this.http = problem.http;
+    this.type = problem.type;
+    this.title = problem.title;
+    this.detail = problem.detail;
+    this.instance = problem.instance;
+    this.stack = problem.stack;
 
-   toJSON(problem: Problem, includeStack = false): ProblemInterface {
+    // add extra keys
+    Object.keys(problem)
+      .filter((el) => !DEFAULT_KEYS.includes(el))
+      .forEach((k) => (this[k] = problem[k]));
+  }
 
-        const { name, message, stack, ...rest } = problem;
+  copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
+    switch (mode) {
+      case COPY_MODE.LEAVE_PROPS:
+        return new Problem(problem, props);
 
-        const jsonObject = {
-            ...rest
+      case COPY_MODE.SKIP_PROPS:
+      default:
+        let keysToBeCopied: string[] = [];
+        for (let key in problem) {
+          if (props.includes(key)) continue;
+          keysToBeCopied.push(key);
         }
+        return new Problem(
+          { type: problem.type, title: problem.title },
+          keysToBeCopied
+        );
+    }
+  }
 
-        if (includeStack)
-            jsonObject.stack = stack;
-
-        return jsonObject;
+  toJSON({ includeStack = false }: ToJsonParamType) {
+    const { stack, ...rest } = this;
 
+    if (includeStack) {
+      return {
+        ...this,
+        stack: this.stack,
+      };
     }
-};
\ No newline at end of file
+
+    return { ...rest };
+  }
+
+  isOfType(type: string) {
+    return this.type === type;
+  }
+
+  async update({ updates }: UpdateProblemParamType) {
+    await Object.keys(updates).forEach((i) => {
+      this[i] = updates[i];
+    });
+  }
```

Comment:
Why `async/await`? It will be always synchronous code:

```suggestion
  update({ updates }: UpdateProblemParamType) {
   Object.keys(updates).forEach((i) => {
      this[i] = updates[i];
    });
  }
```